### PR TITLE
[Group 5] Enable nullable annotations for `Microsoft.Extensions.Configuration.Ini`

### PR DESCRIPTION
**Pull request by maxkoshevoi**
Related to #43605

Annotated according to:
- https://github.com/dotnet/runtime/pull/57434
- https://github.com/dotnet/runtime/pull/57418

### CODE REVIEW

**Comment by maxkoshevoi** [comment]
@eerhardt could you reopen this please?

File: src/libraries/Microsoft.Extensions.Configuration.Ini/ref/Microsoft.Extensions.Configuration.Ini.csproj [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.Ini/ref/Microsoft.Extensions.Configuration.Ini.csproj:2]:
```suggestion
```

File: src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs [hunk 2]
```diff
@@ -23,16 +23,16 @@ public class IniStreamConfigurationProvider : StreamConfigurationProvider
...
-            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            var data = new Dictionary<string, string?>(StringComparer.OrdinalIgnoreCase);
             using (var reader = new StreamReader(stream))
             {
                 string sectionPrefix = string.Empty;
 
                 while (reader.Peek() != -1)
                 {
-                    string rawLine = reader.ReadLine();
+                    string rawLine = reader.ReadLine()!;
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs:35]: (nit) It might be good to add a comment here why this is safe. (I assume it is because Peek returned non `-1`.

File: src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs [hunk 3]
```diff
@@ -23,16 +23,16 @@ public class IniStreamConfigurationProvider : StreamConfigurationProvider
         /// </summary>
         /// <param name="stream">The stream of INI data.</param>
         /// <returns>The <see cref="IDictionary{String, String}"/> which was read from the stream.</returns>
-        public static IDictionary<string, string> Read(Stream stream)
+        public static IDictionary<string, string?> Read(Stream stream)
         {
-            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            var data = new Dictionary<string, string?>(StringComparer.OrdinalIgnoreCase);
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs:28]: When will this dictionary contain `null` values? I don't see it how `null` gets in here.

Comment (maxkoshevoi) [reply src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs:28]: Oh shoot, thanks! This was left from times when I thought, `ReadLine` can return `null`

Comment (maxkoshevoi) [reply src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs:28]: Results from `Read` are assigned to `Data` (that has `IDictionary<string, string?>` type), so `Read` also needs to return `IDictionary<string, string?>`

File: src/libraries/Microsoft.Extensions.Configuration.Json/src/JsonConfigurationProvider.cs [hunk 4]
```diff
@@ -2,7 +2,6 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
-using System.Collections.Generic;
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Configuration.Json/src/JsonConfigurationProvider.cs:5]: Can you revert this in this change? Unnecessary changes should be avoided so we don't clutter source control history.

File: src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs (hunk 3 above)
Comment (eerhardt) [reply src/libraries/Microsoft.Extensions.Configuration.Ini/src/IniStreamConfigurationProvider.cs:28]: Ok, that makes sense. This is `public` API, but its main use case is to create the `Data` dictionary in configuration, which is `string, string?`, so it makes sense for this public API to be consistent.

**Review by eerhardt — APPROVED** [review]

**Comment by eerhardt** [comment]
Failure was an infrastructure issue. https://helix.dot.net/api/2019-06-17/jobs/112777df-f64a-4f39-ab04-c96df3bd783c/workitems/System.Runtime.InteropServices.Tests/console

```
OSError: [Errno 28] No space left on device
```
